test/index.test.js

```diff
@@ -1,9 +1,77 @@
 'use strict';
 
 const assert = require('chai').assert;
+const fs = require('fs');
+const path = require('path');
+
+/**
+ * Read a template file in the "data" directory by filename
+ * @method readConfig
+ * @param  {String}   filename Name of the file to read
+ * @return {Promise}           A promise that resolves to a string, which represents
+ *                             the contents of the file
+ */
+function readConfig(filename) {
+    const filepath = path.resolve(__dirname, 'data', filename);
+
+    return new Promise((resolve, reject) => {
+        fs.readFile(filepath, 'utf8', (err, data) => {
+            if (err) {
+                return reject(err);
+            }
+
+            return resolve(data);
+        });
+    });
+}
 
 describe('index test', () => {
-    it('passes', () => {
-        assert.isTrue(true);
+    let main;
+
+    beforeEach(() => {
+        /* eslint-disable global-require */
+        main = require('../index');
+        /* eslint-enable global-require */
     });
+
+    it('parses a valid yaml', () => {
+        const validFullTemplate = readConfig('valid_full_template.yaml');
+
+        return validFullTemplate
```

Comment:
Tests are documentation for code. Better to explicitly show how this code is called.

```
return main(yamlStr).then(...);
```